### PR TITLE
CODENVY-235: check write access to install directory; code cleanup

### PR DESCRIPTION
**Pull request by dmytro-ndp**
@tolusha: please, review this request.

Signed-off-by: Dmytro Nochevnov <dnochevnov@codenvy.com>

### CODE REVIEW

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh [hunk 1]
```diff
@@ -928,6 +941,15 @@ doCheckAvailableResourcesLocally() {
         exit 1;
     fi
 
+    if [[ ${writeAccessIssueFound} == true ]]; then
+        if [[ -n $INSTALL_DIRECTORY ]]; then
+            println "$(printError "ERROR: Installation directory") $(printImportantLink $INSTALL_DIRECTORY) $(printError "cannot be written to.")"
+        else
+            println $(printError "ERROR: Current directory doesn't have write access.")
```
Comment (tolusha) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: it is better to say: User doesn't have write permissions to directory

**Comment by tolusha** [comment]
New param should be reflected in the doc

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh [hunk 2]
```diff
@@ -914,11 +920,18 @@ doCheckAvailableResourcesLocally() {
...
+        writeAccessIssueFound=true
+        writeAccessStateToDisplay=$(printf "%-44s" && printError "[NOT OK]")
+    fi
+
     println
     println "                MINIMUM        RECOMMENDED     AVAILABLE"
     println "RAM             $minRAMToDisplay $recRAMToDisplay $availableRAMToDisplay $RAMStateToDisplay"
     println "CPU             $minCoresToDisplay $recCoresToDisplay $availableCoresToDisplay $coresStateToDisplay"
     println "Disk Space      $minDiskSpaceToDisplay $recDiskSpaceToDisplay $availableDiskSpaceToDisplay $diskStateToDisplay"
+    println "Write Access    $writeAccessStateToDisplay"
```
Comment (tolusha) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: Access -> Permissions ?

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh [hunk 3]
```diff
@@ -928,6 +941,15 @@ doCheckAvailableResourcesLocally() {
         exit 1;
     fi
 
+    if [[ ${writeAccessIssueFound} == true ]]; then
+        if [[ -n $INSTALL_DIRECTORY ]]; then
+            println "$(printError "ERROR: Installation directory") $(printImportantLink $INSTALL_DIRECTORY) $(printError "cannot be written to.")"
```
Comment (tolusha) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: to -> into

**Comment by tolusha** [comment]
What will be the default directory to install IM into ?

**Comment by TylerJewell** [comment]
The default should be:
<current-directory>/codenvy ... meaning that we create a codenvy directory to hold everything first, and then put codenvy.properties, downloads, etc into that directory.

**Comment by tolusha** [comment]
@dmytro-ndp I don't see of INSTALL_DIRECTORY is used.

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh (hunk 2 above)
Comment (dmytro-ndp) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: That is how Tyler defined the message in description to the https://jira.codenvycorp.com/browse/CODENVY-235
But, I agree - "Write Permission" is more suitable here.

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh (hunk 3 above)
Comment (dmytro-ndp) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: That is how it defined [here](https://jira.codenvycorp.com/browse/CODENVY-235?focusedCommentId=273935&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-273935).

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh (hunk 2 above)
Comment (TylerJewell) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: If you use "Write Permission" then you will have to adjust the columns as the formatting will be altered.

**Comment by dmytro-ndp** [comment]
It will be used in the issue [CODENVY-317](https://jira.codenvycorp.com/browse/CODENVY-235).

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh (hunk 1 above)
Comment (dmytro-ndp) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: I don't think it is good message so as it doesn't say which user doesn't have access to which directory.

**Comment by dmytro-ndp** [comment]
@tolusha
> New param should be reflected in the doc
Which doc do you mean? There is description already [here](https://github.com/codenvy/codenvy/blob/CODENVY-235/cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh#L13).

**Comment by dmytro-ndp** [comment]
@tolusha 
> What will be the default directory to install IM into ?

That is the question for another issue. May be for [this one](https://jira.codenvycorp.com/browse/CODENVY-317).

File: cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh (hunk 2 above)
Comment (dmytro-ndp) [inline cdec/installation-manager-resources/src/main/resources/im-scripts/install-codenvy.sh]: okay

**Comment by tolusha** [comment]
others ok. +1

**Pull request merged**